frontend/src/api/djangoAPI.js

```diff
@@ -1,5 +1,59 @@
 import axios from 'axios';
 
-export default axios.create({
+/**
+ * initialize the client with the base url
+ */
+const http = axios.create({
     baseURL: process.env.REACT_APP_DJANGO_API
-})
\ No newline at end of file
+})
+
+class djangoAPI {
+
+    /**
+     * performs GET request to retrieve full videos list
+     *
+     * @returns {Promise<void>}
+     *          videos list
+     */
+    getAllVideos() {
+        return http.get('videos/');
+    }
+
+    /**
+     * performs GET request to retrieve videos list from searchbar entry
+     * 
+     * @param name 
+     *          searchbar query
+     * @returns {Promise<void>}
+     *          videos list
+     */
+    getVideosByName(name) {
```

Comment:
I think this function name is not very appropriate. 
We are doing a search in the database when requesting videos with a search_query.
The argument is also confusing.